### PR TITLE
Components/DeviceInfo: Replaced `react-responsive-carousel` w/CSS

### LINKED ISSUE #427: Replace carousel lib with CSS
`react-responsive-carousel` is only being used for the [device snapshot images](https://github.com/commaai/connect/blob/master/src/components/DeviceInfo/index.jsx#L376) on mobile. We can implement scrolling with snapping easily using [`scroll-snap-type`](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-type) CSS property.

### PR DESCRIPTION
**Pull request by ugtthis**
**TL;DR**  This replaces carousel lib with CSS which fixes #427 

## Scroll bar always shows (except for Firefox & Safari)
By default `scroll-snap-type` hides scroll bar if user isn't using it, I decided to set it to be always on because pagination circles are not there anymore to indicate that there is another photo to be seen. I also made scroll bar a bit more noticeable by making it taller so users can find out quicker that it is scrollable/swipe-able. I noticed Firefox and Safari does not seem to support the scrollbar to always show or modify its height, but it will still show if you are actively scrolling. Safari works when it's in "responsive mode" but when I simulate it in Xcode the scroll bar goes back to default styling.

## Circle pagination did not seem worth it
Pagination with the little circles could be implemented but I would be adding more JS so I figured for this situation, just using the always on scrollbars would be enough.

## Testing
I tested it on Brave, Chrome, Safari, and Firefox using my laptop. The core functionality of it snapping into place is there for all of them. Firefox and Safari are the browsers that seems to not like the scrollbar being modified as mentioned above. But other than that core functionalities of it snapping works.<br>

## Old vs New

https://github.com/commaai/connect/assets/142481257/bc07e732-14a3-45df-8072-38680b181c9a

## Testing through different browsers

https://github.com/commaai/connect/assets/142481257/20594ee1-005b-4eaa-8863-41adb6dd74cb

### CODE REVIEW

**Comment by ugtthis** [comment]
Got a little worried that I did something wrong because of the really big diff number, and looked into it. To my understanding this is happening because of the removal of the `react-responsive-carousel` library which means `pnpm` then recalculates the whole dependency tree, which could mean removing, updating or adding new dependencies because the `react-responsive-carousel` lib is not there anymore

File: src/components/DeviceInfo/index.jsx [hunk 1]
```diff
@@ -147,6 +147,31 @@ const styles = (theme) => ({
       },
     },
   },
+  // IF on mobile
```
Comment (adeebshihadeh) [inline src/components/DeviceInfo/index.jsx]: can you remove all these added comments? you generally don't want comments that rewrite the code into english, the code should be readable on its own

Comment (ugtthis) [reply src/components/DeviceInfo/index.jsx]: Saw you already took off the comments in the cleanup commit. Thanks for doing that and for the feedback, will note that for next time
